File: api/Shop/Model/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using Shop.Model.Relation;

namespace Shop.Model;

public class Order
{
    [Key] public uint ID { get; set; }
    public string Status { get; set; }
    [ForeignKey("Customer")] public uint CustomerID { get; set; }
    public ICollection<OrderProduct> Products { get; set; }
}

File: api/shop/model/order.go
package model

import "time"

type Order struct {
	ID           uint    `gorm:"primaryKey" json:"id"`
	CustomerID   uint    `json:"customerID"`
	FullPrice    float32 `json:"fullPrice"`
	Status       string  `json:"status"`
	CreatedAt    time.Time
	ExtraComment string `json:"extraComment"`
}
